### PR TITLE
SipHash: Fix CTRL API for the digest size.

### LINKED ISSUE #14370: SipHash via EVP_PKEY has lost support for EVP_PKEY_CTRL_SET_DIGEST_SIZE
In 1.1.1 using SipHash via EVP_PKEY you could use the ctrl `EVP_PKEY_CTRL_SET_DIGEST_SIZE` or the ctrl string "digestsize". In 3.0, this is no longer possible. The parameter seems to have been renamed to "size". 

Running the 1.1.1 tests against 3.0 I experienced this error:

````
       # ../../test/recipes/30-test_evp_data/evpmac.txt:133: Source of above error; unexpected error EVPPKEYCTXCTRL_ERROR
        # 804B055BFB7F0000:error:03000093:digital envelope routines:default_fixup_args:command not supported:crypto/evp/ctrl_params_translate.c:582:[action:2, state:4] name=digestsize, value=8
````

### PR DESCRIPTION
**Pull request by paulidale**
Adds a mapping from "digestsize" to "size" for setting a param.

Fixes #14370

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: crypto/evp/ctrl_params_translate.c [hunk 1]
```diff
@@ -2093,6 +2093,14 @@ static const struct translation_st evp_pkey_ctx_translations[] = {
       EVP_PKEY_CTRL_RSA_KEYGEN_PRIMES, "rsa_keygen_primes", NULL,
       OSSL_PKEY_PARAM_RSA_PRIMES, OSSL_PARAM_UNSIGNED_INTEGER, NULL },
 
+    /*-
+     * SipHash
+     * ======
+     */
+    { SET, -1, -1, -1,
```
Comment (slontis) [inline crypto/evp/ctrl_params_translate.c:2100]: Looks ok to me - maybe @levitte needs to confirm this one//

Comment (levitte) [reply crypto/evp/ctrl_params_translate.c:2100]: The third -1 should probably be `EVP_PKEY_OP_TYPE_SIG`...  remember that this is for the old way of making MACs, via the PKEY signature API

Comment (paulidale) [reply crypto/evp/ctrl_params_translate.c:2100]: I wasn't sure what was appropriate there and _-1_ worked :)

Comment (paulidale) [reply crypto/evp/ctrl_params_translate.c]: Fix submitted.

**Comment by paulidale** [comment]
Ping for review?

**Review by slontis — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the reviews.

**Pull request closed without merging**